apcd-cms/src/apps/submissions/views.py

```diff
@@ -28,11 +28,27 @@ def get_context_data(self, *args, **kwargs):
 
         submission_content = get_user_submissions_and_logs(user)
 
+        filter = self.request.GET.get('filter')
+        dateSort = self.request.GET.get('sort')
+
+        def getDate(row):
+            date = row['received_timestamp']
+            return parser.parse(date) if date is not None else parser.parse('1-1')
+
+        if dateSort is not None:
+            context['selected_sort'] = dateSort
+            submission_content = sorted(submission_content, key=lambda row:getDate(row), reverse=(dateSort == 'newDate'))
+
         try:
             page_num = int(self.request.GET.get('page'))
         except:
             page_num = 1
 
+        context['selected_filter'] = None
+        if filter is not None and filter != 'All':
+            context['selected_filter'] = filter
+            submission_content = table_filter(filter, submission_content, 'status')
+
         limit = 50
         offset = limit * (page_num - 1)
 
```

Comment:
Line 57 I think the formatting is only applying to the first page because of the [offset:offset + limit] specification for this iteration. I'm not seeing the formatted submission content beyond the first page. Do you think this could be the culprit? I tested and got all the fields to format without this, but it wasn't solving the problem of date ordering.